clients/drcachesim/simulator/caching_device.cpp

```diff
@@ -240,8 +240,8 @@ caching_device_t::request(const memref_t &memref_in)
 
         // Issue a hardware prefetch, if any, before we remember the last tag,
         // so we remember this line and not the prefetched line.
-        if (missed && !type_is_prefetch(memref.data.type) && prefetcher_ != nullptr)
-            prefetcher_->prefetch(this, memref);
```

Comment:
I understand the value in moving the "missed" check into prefetcher->prefetch(), but the type_is_prefetch() check seems a little less desirable to move.  The simulation relies on this to avoid an infinite loop of prefetching, but I'm not sure that's obvious enough to prefetcher implementors.  Is there a use-case for a prefetcher seeing prefetches?

Reply:
Use case was a bit speculative. Reverted this part for now and we can revisit it if it becomes necessary.